### PR TITLE
check for inputProps value when setting query in Omnibar

### LINKED ISSUE #1380: [Labs] Controlled usage of Omnibox component
### Bug report

- __Package version(s)__: @blueprintjs/labs: ^0.4.0
- __Browser and OS versions__: Chrome Version 59.0.3071.115, MAC OS Sierra 10.12.5

#### Steps to reproduce

1. Override the `inputProps` property with `onChange` and `value` properties.
2. Type text into omnibox input
3. No InputGroup or noResults JSX element is rendered

#### Actual behavior

When attempting to override the `inputProps` for the `onChange` and `value` properties, the component becomes unresponsive. It looks like the state variable `query` in the Omnibox component is not getting updated when switching to controlled usage.

#### Expected behavior

Override the `inputProps` should still trigger a render of the resulting InputGroup or no result JSX element.

### PR DESCRIPTION
**Pull request by dan-katz**
#### Checklist

- [x] [Enable CircleCI for your fork](https://circleci.com/add-projects)

#### Changes proposed in this pull request:

- In Omnibar, check if there is a value from inputProps when setting the query in componentWillReceiveProps. If there is one then use it and if not use this.state.query

#### Reviewers should focus on:

- Whether the Omnibar component works as expected

### CODE REVIEW

**Comment by dan-katz** [comment]
<h3>check for inputProps value when setting query</h3>

Preview: <strong><a href='https://47-127456300-gh.circle-artifacts.com/0/home/circleci/project/packages/docs-app/dist/index.html' target='_blank'>documentation</a> | <a href='https://47-127456300-gh.circle-artifacts.com/0/home/circleci/project/packages/landing-app/dist/index.html' target='_blank'>landing</a> | <a href='https://47-127456300-gh.circle-artifacts.com/0/home/circleci/project/packages/table-dev-app/dist/index.html' target='_blank'>table</a></strong>

**Comment by dan-katz** [comment]
I'm going to close this and resubmit it to the release/2.x branch

**Comment by giladgray** [comment]
@dan-katz please don't do that. i do not want to ship any more 2.x releases. 3.0 or bust

File: packages/select/src/components/omnibar/omnibar.tsx [hunk 1]
```diff
@@ -94,9 +94,14 @@ export class Omnibar<T> extends React.PureComponent<IOmnibarProps<T>, IOmnibarSt
         const { isOpen } = nextProps;
         const canClearQuery = !this.props.isOpen && isOpen && this.props.resetOnSelect;
 
+        const query =
+            nextProps.inputProps && nextProps.inputProps.value
+                ? (nextProps.inputProps.value as string)
+                : this.state.query;
```
Comment (giladgray) [inline packages/select/src/components/omnibar/omnibar.tsx]: while this is the right idea, it's not enough to actually solve this issue. you've fixed `Omnibar` but not `Select` or any of the others. i'm actually working on controlled support _right now_ so perhaps just hang tight?

**Pull request closed without merging**